forte/data/ontology/top.py

```diff
@@ -340,14 +333,14 @@ def set_parent(self, parent: SubEntry):  # type: ignore
         Returns:
 
         """
-        if not isinstance(parent, self.ParentType):
+        if not isinstance(parent, Entry):
```

Comment:
I had to change this because the gpt2 example was throwing the following error 

```Traceback (most recent call last):
  File "multipack_pipeline_gpt2.py", line 20, in <module>
    multipack = pl.process_one(input_dir)
  File "/home/centos/new_space/forte/forte/base_pipeline.py", line 235, in process_one
    results = [p for p in self.process_packs(iter(first_pack))]
  File "/home/centos/new_space/forte/forte/base_pipeline.py", line 235, in <listcomp>
    results = [p for p in self.process_packs(iter(first_pack))]
  File "/home/centos/new_space/forte/forte/base_pipeline.py", line 277, in process_packs
    self._processors[job.step_num].process(c_pack)
  File "/home/centos/new_space/forte/forte/processors/base/base_processor.py", line 53, in process
    self._process(input_pack)
  File "/home/centos/new_space/forte/forte/processors/base/batch_processor.py", line 104, in _process
    self.pack_all(pred)
  File "/home/centos/new_space/forte/forte/processors/base/batch_processor.py", line 139, in pack_all
    self.pack(self.batcher.data_pack_pool[i], output_dict_i)
  File "/home/centos/new_space/forte/forte/processors/text_generation_processor.py", line 175, in pack
    data_pack.subentry(self.output_pack_name, sent))
  File "/home/centos/new_space/forte/forte/data/ontology/top.py", line 310, in __init__
    super().__init__(pack, parent, child)
  File "/home/centos/new_space/forte/forte/data/ontology/core.py", line 169, in __init__
    self.set_parent(parent)
  File "/home/centos/new_space/forte/forte/data/ontology/top.py", line 336, in set_parent
    if not isinstance(parent, self.ParentType):
AttributeError: 'MultiPackLink' object has no attribute 'ParentType'
```

Essentially, `subentry` method is failing whenever I use `self.ParentType`. Similar reasons for `self.ChildType`